_validate/validate.py

```diff
@@ -5,75 +5,116 @@
 # For more details see: https://www.gnu.org/licenses/gpl-2.0.html
 
 import argparse
+import enum
 import os
-import sys
 import tempfile
+import typing
 import zipfile
 import json
 import urllib.request
 from jsonschema import validate, exceptions
-
-sys.path.append(os.path.dirname(__file__))
-import sha256
-from addonManifest import AddonManifest
-del sys.path[-1]
+from . import sha256
+from .addonManifest import AddonManifest
 
 
 JSON_SCHEMA = os.path.join(os.path.dirname(__file__), "addonVersion_schema.json")
-DOWNLOAD_BLOCK_SIZE = 8192 # 8 kb
 TEMP_DIR = tempfile.gettempdir()
 
-def getAddonMetadata(filename):
+JsonObjT = typing.Dict[str, typing.Any]
+
+
+@enum.unique
+class ValidationErrorMessage(enum.Enum):
+	URL_MISSING_HTTPS = "Add-on download url must start with https://"
+	URL_MISSING_ADDON_EXT = "Add-on download url must end with .nvda-addon"
+	URL_DOWNLOAD_ERROR = "Download of addon failed"
+	CHECKSUM_FAILURE = "Sha256 of .nvda-addon at URL is: {}"
+	NAME = "Submission 'displayName' must be set to '{}' in json file. Instead got: '{}'"
+	DESC = "Submission 'description' must be set to '{}' in json file. Instead got: '{}'"
+	HOMEPAGE = "Submission 'homepage' must be set to '{}' in json file"
+	SUBMISSION_DIR_ADDON_NAME = "Submitted json file must be placed in {} folder."
+	SUBMISSION_DIR_ADDON_VER = "Submitted json file should be named {}.json"
+
+
+ErrorGenerator = typing.Generator[str, None, None]
```

Comment:
Because the behaviour between normal Python errors and validation errors is quite different, I think it would be good to make that distinction clear in naming these objects. Maybe even using the term "Validation Warning/Failure" instead of "Validation Error"

```suggestion
ValidationErrorGenerator = typing.Generator[str, None, None]
```